wakuv2/waku.go

```diff
@@ -1007,6 +1014,38 @@ func (w *Waku) broadcast() {
 	}
 }
 
+func (w *Waku) checkIfMessagesStored() {
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-w.ctx.Done():
+			w.logger.Debug("Stop the look for message stored check")
+			return
+		case <-ticker.C:
+			w.logger.Debug("Running loop for messages stored check")
+			w.logger.Debug("Send Message IDs", zap.Any("sendMsgIDs", w.sendMsgIDs))
+			w.sendMsgIDsMu.Lock()
```

Comment:
It will be problematic if we lock here and then loop through and process all messages and do store query to fetch them, sender will be blocked for a long time and will slow down over time. 

Store query should be kept outside this lock somehow or use some fine-grain locking if needed. 

Or can we think of some other alternative datastructure to achieve this so that we don't lock the whole messages datastructure? Maybe use some sort of time based datastructure so that only the reference that is currently worked on i.e older than 5 secs get locked.